optimum/intel/openvino/modeling_decoder.py

```diff
@@ -473,3 +483,8 @@ def _convert_to_standard_cache(
     def can_generate(self):
         """Returns True to validate the check that the model using `GenerationMixin.generate()` can indeed generate."""
         return True
+
+
+# bf16 may be in used when quantization, disable pastkv lower precision optimization
+class OVModelForCausalLMDisablePastKVOpt(OVModelForCausalLM):
```

Comment:
Can you explain why this is needed? 
If it is necessary to disable this for quantized models, can it be done without requiring a separate class? For example with a `disable_pkv_optimizations` parameter to the standard class? It is very unexpected to have to use a different model class for quantizided models. I'm also wondering if we can detect this case automatically.

Reply:
Quantization will directly access the `self.model.model`, `self.model.request` and assign quantized model to `self.model.model`. If the optimization is not disabled, `self.model.model` will stand the modified model which nncf will not recognize it. Even if we save the modified model to `self.model.model_new`, keep the original model in `self.model.model`, `_quantize_ovcausallm` will collect inputs/outputs which may contain bf16 data and nncf will pass collected inputs/outputs to `self.model.model`, it will fail because datatype is mismatched.
Simply disabling the optimization can avoid the difficult situation when doing quantization, and it should be reasonable, quantized model should be mixed f32 and int8, but bf16 and int8.
If new class is not acceptable, may I use `ov_config['EXECUTION_MODE_HINT']='ACCURACY'` to disable the optimization when doing quantization?

Reply:
Can you remove the new class and change `_quantize_ovcausallm` accordingly? Let me know if you need any help here.

Reply:
nncf's input parameter `model` should be in type `openvino.runtime.ie_api.Model`, and the modified model has the type `openvino._pyopenvino.Model`. It means the original unmodified model should be used, so I'd suggest to use `ov_config['EXECUTION_MODE_HINT']='ACCURACY'` disable the modified model since it's no use when doing quantization, and the new class will be removed also.